src/math/range.h

```diff
@@ -5,19 +5,22 @@
 
 #include "base/serialiser.h"
 #include "math/matrix4.h"
+#include <optional>
 
 // Range
 class Range : public Serialisable<>
 {
     public:
-    Range(double minimum = 0.0, double maximum = 0.0);
+    Range(std::optional<double> minimum = std::nullopt, std::optional<double> maximum = std::nullopt);
 
     /*
      * Data
      */
     private:
     // Minimum and maximum for range
-    double minimum_, maximum_;
+    std::optional<double> minimum_, maximum_;
+    // Set range
+    void set(std::optional<double> minimum, std::optional<double> maximum);
```

Comment:
Unwritten code convention suggests that member variables are separate from functions for clarity:
```suggestion
    std::optional<double> minimum_, maximum_;

    private:
    // Set range
    void set(std::optional<double> minimum, std::optional<double> maximum);
```